### PR TITLE
Add Intuit provider

### PR DESCRIPTION
**Pull request by epotter2297**
This PR adds [Intuit OAuth](https://developer.intuit.com/app/developer/qbo/docs/develop/authentication-and-authorization/oauth-2.0) provider ([#96](https://github.com/pilcrowOnPaper/arctic/issues/96))

### CODE REVIEW

File: src/providers/intuit.ts [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+		const scopes = options?.scopes ?? [];
+		return await this.client.createAuthorizationURL({
+			state,
+			scopes: [...scopes, "openid"]
+		});
+	}
+
+	public async validateAuthorizationCode(code: string): Promise<IntuitTokens> {
+		const result = await this.client.validateAuthorizationCode<TokenResponseBody>(code, {
+			authenticateWith: "request_body",
```
Comment (pilcrowonpaper) [inline src/providers/intuit.ts]: The credentials should be send within the authorization header, so this should be `http_basic_auth`. That's the default in Oslo, you can just remove this line.

https://developer.intuit.com/app/developer/qbo/docs/develop/authentication-and-authorization/oauth-2.0#exchange-code-for-tokens

File: src/providers/intuit.ts [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+			refreshToken: result.refresh_token,
+			refreshTokenExpiresAt: createDate(new TimeSpan(result.x_refresh_token_expires_in, "s")),
+			idToken: result.id_token
+		};
+		return tokens;
+	}
+
+	public async refreshAccessToken(accessToken: string): Promise<IntuitTokens> {
+		const result = await this.client.refreshAccessToken<TokenResponseBody>(accessToken, {
+			authenticateWith: "request_body",
```
Comment (pilcrowonpaper) [inline src/providers/intuit.ts]: ditto

**Review by pilcrowonpaper — CHANGES_REQUESTED** [review]
Can you also update the providers list in the README?

**Comment by epotter2297** [comment]
Those updates should all be done now :)

**Review by pilcrowonpaper — APPROVED** [review]

**Comment by pilcrowonpaper** [comment]
Thanks!

**Pull request merged**